core/src/main/java/com/scalar/db/storage/jdbc/JdbcDatabase.java

```diff
@@ -93,7 +93,7 @@ public Optional<Result> get(Get get) throws ExecutionException {
 
   @Override
   public Scanner scan(Scan scan) throws ExecutionException {
-    scan = copyAndSetTargetToIfNot(scan);
+    scan = (Scan) copyAndSetTargetToIfNot(scan);
```

Comment:
I guess that the caller of `copyAndSetTargetToIfNot(ScanInterface)` expects the same return type as the parameter type (e.g. expects `Scan` when passing `Scan` instance.) If so, how about keeping the existing `Scan copyAndSetTargetToIfNot(Scan)` and adding `RelationalScan copyAndSetTargetToIfNot(RelationalScan)` later so that we can remove this class cast?

Reply:
I'm not sure why I chose the cast rather than adding the relational one, but now I think your suggestion is far better. I will fix it (including other comments). Thank you!

Reply:
Fixed in d6da904. Thank you!